compiler/test/stdlib/path.test.gr

```diff
@@ -0,0 +1,165 @@
+import Path from "path"
+import Result from "result"
+import Option from "option"
+import List from "list"
+
+// test that a variety of properties hold for valid relative/absolute paths
+let relTests = [
+  (
+    "./dir/file.txt",
+    "./dir",
+    "./dir/file.txt",
+    Some("file.txt"),
+    Some("file"),
+    Some(".txt"),
+  ),
+  ("file", ".", "./file", Some("file"), Some("file"), None),
+  ("dir/../../", "../..", "..", None, None, None),
+  (".git", ".", "./.git", Some(".git"), Some(".git"), None),
+  (".", "..", ".", None, None, None),
+  ("..", "../..", "..", None, None, None),
+  ("../../file", "../..", "../../file", Some("file"), Some("file"), None),
+  ("./", "..", ".", None, None, None),
+  (
+    "dir//file.txt///",
+    "dir",
+    "./dir/file.txt",
+    Some("file.txt"),
+    Some("file"),
+    Some(".txt"),
+  ),
+  (
+    ".\\dir\\file.txt",
+    ".",
+    "./.\\dir\\file.txt",
+    Some(".\\dir\\file.txt"),
+    Some(".\\dir\\file"),
+    Some(".txt"),
```

Comment:
Can we get a double extension test? I'm thinking something like `webpack.config.js`

Reply:
I could've sworn I included a test like this on a `.tar.gz` gzip extension, but I guess I removed it accidentally.